mobile/src/navigation/app/app.navigation.tsx

```diff
@@ -2,37 +2,61 @@ import { createDrawerNavigator } from '@react-navigation/drawer';
 import React, { FC } from 'react';
 
 import { AppScreenName } from '~/common/enums/enums';
-import { AppNavigationParamList } from '~/common/types/types';
-import { Billing } from '~/components/billing/billing';
-import { Courses } from '~/components/courses/courses';
-import { Mentors } from '~/components/mentors/mentors';
-import { MyEducation } from '~/components/my-education/my-education';
-import { Overview } from '~/components/overview/overview';
-import { Settings } from '~/components/setting/setting';
-import { UAM } from '~/components/uam/uam';
-
-import { SCREEN_OPTIONS } from './common/constants';
+import {
+  AppNavigationParamList,
+  DrawerNavigationItem,
+  PermissionItem,
+} from '~/common/types/types';
+import { useAppSelector, useMemo } from '~/hooks/hooks';
+
+import { NAVIGATION_ITEMS, SCREEN_OPTIONS } from './common/constants';
 import { DrawerContent } from './components/components';
 
 const Drawer = createDrawerNavigator<AppNavigationParamList>();
 
 const App: FC = () => {
+  const permissions = useAppSelector((state) => state.auth.user?.permissions);
+
+  const getAllowedScreens = (
+    screens: DrawerNavigationItem[],
+    userPermissions: PermissionItem[] | undefined,
```

Comment:
We should have there PermissionKey not PermissionItem

Reply:
@Liza-Veis 
But UserWithPermissions has field permissions: PermissionsGetAllItemResponseDto[], not permissions: PermissionKey[];

Reply:
I think I understand

Reply:
We have to use the field key from PermissionsGetAllItemResponseDto

But why is there a string type, not PermissionKey